### PR TITLE
Improve process execution speed

### PR DESCRIPTION
**Pull request by caleeli**
## Issue & Reproduction Steps
- Calculate next user is slow and not required for Service and Script Tasks
- Add _user and _request are slow in Task Activation and should not be done in Repository anymore
- There are places where we are using Eloquent relationships to get Objects (like processRequest, processVersion, processRequestTokens) that were previously loaded this adds a lot of payload when running a process
- `function updateCatchEvents` is used during process execution and it is taking considerable time, needs to be improved.

## Solution
- The user_id in Service and Script tasks are always null, there is no need to calculate its value
- To keep the backward compatibility _user and _request are still being set in Task Activation, but includes some speed improvements
- Reduce the use of eloquent relationships (do not run and additional query to load already loaded objects) to speed up the process
- Reduce the use of eloquent relationships and save the changes only once

## How to Test

- Create a process with DataSources.
- Create a process with User Tasks.
- Create a process with intermediate catch events.
- Run them.
- The processes should work as expected.

[Test Catch Events.zip](https://github.com/ProcessMaker/processmaker/files/8287121/Test.Catch.Events.zip)

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-5776

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

### CODE REVIEW

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://bugfix-improve_update_catch_events-processmaker.ci.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://bugfix-improve_update_catch_events-processmaker.ci.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://bugfix-improve_update_catch_events-processmaker.ci.processmaker.net

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://bugfix-improve_update_catch_events-processmaker.ci.processmaker.net

**Review by sanjacornelius — APPROVED** [review]

**Review by danloa — APPROVED** [review]

**Comment by caleeli** [comment]
This PR also contributes to: https://processmaker.atlassian.net/browse/FOUR-5452

File: ProcessMaker/Models/CallActivity.php [hunk 1]
```diff
@@ -227,7 +227,7 @@ public function addToken(ExecutionInstanceInterface $instance, TokenInterface $t
      */
     private function syncronizeInstances(ExecutionInstanceInterface $instance, ExecutionInstanceInterface $currentInstance)
     {
-        if ($instance->process->id !== $currentInstance->process->id) {
+        if ($instance->getProcess()->getOwnerDocument()->getModel()->id !== $currentInstance->getProcess()->getOwnerDocument()->getModel()->id) {
```
Comment (ryancooley) [inline ProcessMaker/Models/CallActivity.php:230]: This is a long chain. Is it possible any of these could fail and cause a `Call to a member function on null` or `Trying to get a property of non object` error?

Comment (caleeli) [reply ProcessMaker/Models/CallActivity.php:230]: @ryancooley thanks for the comment,
`getProcess()` could be null if $instance is created without a process or if the bpmn process ID is invalid.
if getProcess() is valid: `getProcess()->getOwnerDocument()->getModel()` will always hava a valid reference.
I will add a validation

**Comment by caleeli** [comment]
@ryancooley reviewing in deep `getProcess()` could not be null, neither `getProcess()->getOwnerDocument()` if that's the case, it should trigger an error when the request is loaded.
Added an check the cause `getModel()`=null

**Comment by pmkerberos** [comment]
QA server was successfully deployed https://bugfix-improve_update_catch_events-processmaker.ci.processmaker.net

**Review by ryancooley — APPROVED** [review]

**Pull request merged**